### ISSUE TITLE
Form submissions do NOT return to form - some kind of internal error in Mautic

### ISSUE THREAD

**Issue by pitus** [mautic/mautic#315]
Here's a new one we're trying to figure out as we were hoping to start using Mautic this week, yet things that used to work stopped and it's impossible to figure this out ... Perhaps it's 1.0.3 upgrade but it seems the problem started a day before the upgrade...

What's happening is this:
Form has 2 actions: Add a Lead and Add it to a List, then return to Form with a confirmation message
When submitted, Mautic's `/form/submit?formId=5` page is shown blank and nothing else happens visually.
All actions are carried out (the new Lead appears in Lead list) but there's no redirection back to form happening.

I suspect that after executing these two actions, there's some kind of an internal error that's not trapped by Mautic or maybe it is trapped, but prevents the flow from reaching the Redirect to Form code so no redirection happens.

Reloading that URL causes the same form data to be posted again and this time we're redirected back to the form page - so the Form processing on the second POST is somewhat different, no longer breaks and completes the redirect.

If we take out the Add to List action and leave only Create Lead action, then it works fine so adding to the List might be the indirect cause of this behaviour.

I can't take it any further than this as mautic error log file is totally useless - see #314 - and the only errors in it are related to a **double-IP address** failing an INSERT into `audit_log.ip_address` but my IP is singular and fits the sql field so I doubt these are related.

This is so frustrating - we were ready to launch our campaign and hoping that 1.0.3 will solve some of the problems we've ran into and instead, we can't get past a simple form submission anymore...

**Comment by alanhartless** [comment]
Hi Pitus,

Can you check your server's error log to see if it hit a PHP Fatal error rather than an exception? Exceptions are currently logged to Mautic's log but if there was a fatal error, it likely went to your server's error log instead. (Looking into if we can get that routed to our log as well so that it's all in one place).

I replied regarding your double IP issue in #314.

Thanks,
Alan

**Comment by pitus** [comment]
`PHP Fatal error:  Cannot use object of type Mautic\LeadBundle\Entity\Lead as array in /www_dev/httpdocs/PROD/mautic/app/bundles/FormBundle/Model/SubmissionModel.php on line 281`

**Comment by alanhartless** [comment]
Thanks Pitus.

Okay so same as #311.

Can you test this change:

/app/bundles/LeadBundle/Model/LeadModel.php line 434, change it to

```
        if (!$returnTracking && $this->systemCurrentLead) {
```

What form submit actions do you have configured?

Thanks,
Alan

**Comment by pitus** [comment]
Seems to be working now.

**Comment by alanhartless** [comment]
There was a fix elsewhere related to setSystemLead. I wanted to say that it could have been related to existing tracking cookies or existing leads vs new leads but if you say you cleared cookies, etc first, then that's not it.  But the change I suggested should work to force the cookie tracked lead if requested.  I'll PR it and will get it out as a hotfix.

Closing this as a duplicate to #311 so let's continue the conversation there if needed.
